numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1823,3 +1825,193 @@ mapiter_@name@(PyArrayMapIterObject *mit)
 }
 
 /**end repeat**/
+
+
+
+#define ptr_assignment1(ptr1, val1, stride1) *ptr1 = val1; ptr1 += stride1; 
+#define ptr_assignment2(ptr1, val1, stride1, ptr2, val2, stride2) ptr_assignment1(ptr1, val1, stride1) *ptr2 = val2; ptr2 += stride2; 
+
+#define loop_ptr_assignment(start, end, ptr_assignment) \
+    for (npy_intp i=start; i<end; ++i) { \
+        ptr_assignment \
+    }
+
+/**begin repeat
+ * 
+ * #dtype = npy_bool, npy_byte, npy_byte, npy_uint16, npy_int16, npy_uint32, npy_int32, npy_uint64, npy_int64, npy_float, npy_double#
+ * #name = bool, u8, i8, u16, i16, u32, i32, u64, i64, f32, f64#
+ */
+
+/**begin repeat1
+ *
+ * #layout=C,F#
+ *
+ */
+
+void nonzero_idxs_1D_@layout@_@name@(@dtype@ * data, npy_intp* idxs, npy_intp* shape, npy_intp* strides, npy_intp nonzero_count) {
+    npy_intp stride = strides[0]; 
+    npy_intp added_count = 0; 
+    npy_intp a = 0; 
+    while (added_count < nonzero_count) { 
+        *idxs = a;  
+        bool to_increment = ((bool) *data); 
+        idxs += ((int) to_increment); 
+        added_count += ((int) to_increment); 
+        data = (@dtype@ *) (((char*) data) + stride); 
```

Comment:
Data is typed `+=` will work just fine.  If this is c-continguous, `data++;` will work. But I think it is good if this is triggered for all 1-D arrays.